src/app/static/view-question/view-question.component.ts

```diff
@@ -18,63 +18,28 @@ import { AuthService } from 'src/app/services/auth/auth.service';
   templateUrl: './view-question.component.html',
   styleUrls: ['./view-question.component.css'],
 })
-export class ViewQuestionComponent implements OnInit {
+export class ViewQuestionComponent implements OnInit, OnChanges {
   /** For now, ViewQuestionComponent is tightly coupled with the QuestionService
    * and only displays the question whose id is saved in that service.
    * This should be a reusable component for displaying arbitrary questions, but
    * refactoring will take some doing and this works.  So selectedQuestion is
    * appropriate -- it will only ever be the question selected by the user elsewhere
    * on the site.
    */
-  @Output() selectedQuestion: Question;
+  @Input() selectedQuestion: Question;
   responses: Response[];
-  highlightedResponse: any;
-  questionResponses: any;
-  totalMsgCount: number = 0; // count the messages of the responses. initialize to 0;
-
 
   constructor(
     private questionService: QuestionService,
     private authService: AuthService
   ) { }
 
   ngOnInit() {
-    this.refreshPage();
+    this.ngOnChanges(null);
   }
 
-  refreshPage() {
-    // Getting the selected question ID from the QuestionService that was set in the PreviewQuestionComponent
-    const selectedQuestionId = this.questionService.getQuestionId();
-
-    this.questionService.getQuestionById(selectedQuestionId).subscribe(question => {
-      this.selectedQuestion = question;
-    });
-
-    // Retrieving the highlighted response for a specified question
-    this.questionService
-      .getQuestionById(selectedQuestionId)
-      .subscribe(result => {
-        if (result.highlightedResponseId != null) {
-          const responsesByQuestion: any = result.responses; // Gets the objects in 'responses' array in the questions JSON
-          // tslint:disable-next-line: forin
-          for (const r in responsesByQuestion) {
-            const responseId: any = responsesByQuestion[r].id; // Sets the response ID
-            if (responseId === result.highlightedResponseId) {
-              // Checks to see if response ID is equal to the highlighted response ID
-              this.highlightedResponse = responsesByQuestion[r];
-            }
-          }
-        } else {
-          console.log('No highlighted response');
-        }
-      });
```

Comment:
While this logic could use some work, this component might be the right place for actually getting the highlighted response.  Let's put a pin in that functionality and come back to it after the larger rework is done.